contracts/PoseidonHasher.sol

```diff
@@ -1,533 +1,891 @@
+	uint256 constant C106 = 6918585327145564636398173845411579411526758237572034236476079610890705810764;
+	uint256 constant C107 = 14158163500813182062258176233162498241310167509137716527054939926126453647182;
+	uint256 constant C108 = 4164602626597695668474100217150111342272610479949122406544277384862187287433;
+	uint256 constant C109 = 12146526846507496913615390662823936206892812880963914267275606265272996025304;
+	uint256 constant C110 = 10153527926900017763244212043512822363696541810586522108597162891799345289938;
+	uint256 constant C111 = 13564663485965299104296214940873270349072051793008946663855767889066202733588;
+	uint256 constant C112 = 5612449256997576125867742696783020582952387615430650198777254717398552960096;
+	uint256 constant C113 = 12151885480032032868507892738683067544172874895736290365318623681886999930120;
+	uint256 constant C114 = 380452237704664384810613424095477896605414037288009963200982915188629772177;
+	uint256 constant C115 = 9067557551252570188533509616805287919563636482030947363841198066124642069518;
+	uint256 constant C116 = 21280306817619711661335268484199763923870315733198162896599997188206277056900;
+	uint256 constant C117 = 5567165819557297006750252582140767993422097822227408837378089569369734876257;
+	uint256 constant C118 = 10411936321072105429908396649383171465939606386380071222095155850987201580137;
+	uint256 constant C119 = 21338390051413922944780864872652000187403217966653363270851298678606449622266;
+	uint256 constant C120 = 12156296560457833712186127325312904760045212412680904475497938949653569234473;
+	uint256 constant C121 = 4271647814574748734312113971565139132510281260328947438246615707172526380757;
+	uint256 constant C122 = 9061738206062369647211128232833114177054715885442782773131292534862178874950;
+	uint256 constant C123 = 10134551893627587797380445583959894183158393780166496661696555422178052339133;
+	uint256 constant C124 = 8932270237664043612366044102088319242789325050842783721780970129656616386103;
+	uint256 constant C125 = 3339412934966886386194449782756711637636784424032779155216609410591712750636;
+	uint256 constant C126 = 9704903972004596791086522314847373103670545861209569267884026709445485704400;
+	uint256 constant C127 = 17467570179597572575614276429760169990940929887711661192333523245667228809456;
+	uint256 constant M00 = 2910766817845651019878574839501801340070030115151021261302834310722729507541;
+	uint256 constant M01 = 19727366863391167538122140361473584127147630672623100827934084310230022599144;
+	uint256 constant M10 = 5776684794125549462448597414050232243778680302179439492664047328281728356345;
+	uint256 constant M11 = 8348174920934122550483593999453880006756108121341067172388445916328941978568;
+
+	function hash(uint256 input) external pure override returns (uint256 result) {
 		return _hash(input);
 	}
 
-	function _hash(uint256[2] memory input) internal pure returns (uint256 result) {
+	function _hash(uint256 input) internal pure returns (uint256 result) {
```

Comment:
If the hash algorithm internals is inspired by another repo, a link to that repo would be great.